pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -4096,8 +4096,19 @@ func readRow(vars []interface{}, columnNames []string) ([]string, map[string]str
 				groupBy = append(groupBy, fmt.Sprintf("%v", reflect.ValueOf(v).Elem().Int()))
 				groupAttributes[colName] = fmt.Sprintf("%v", reflect.ValueOf(v).Elem().Int())
 			}
+		case *bool:
+			if _, ok := constants.ReservedColumnTargetAliases[colName]; ok {
+				if *v {
+					point.Value = 1
+				} else {
+					point.Value = 0
+				}
```

Comment:
How can a boolean be target column?

Reply:
When we groupBy over a boolean column.

Reply:
Aggregation on the attributes such as sum, max etc are the target columns. Group by is only for aggregation among the groups but group by column itself is not target column.

Reply:
Understood that you are talking about the reserved target column. Pushed the changes.